source/extensions/upstreams/http/generic/config.cc

```diff
@@ -28,8 +31,13 @@ Router::GenericConnPoolPtr GenericGenericConnPoolFactory::createGenericConnPool(
         std::make_unique<Upstreams::Http::Tcp::TcpConnPool>(thread_local_cluster, route_entry, ctx);
     return (conn_pool->valid() ? std::move(conn_pool) : nullptr);
   case UpstreamProtocol::UDP:
+#ifdef ENVOY_ENABLE_QUIC
     conn_pool = std::make_unique<Upstreams::Http::Udp::UdpConnPool>(thread_local_cluster, ctx);
```

Comment:
This surprises me slightly. Maybe we need to rethink/rename this class. But "UdpConnPool" doesn't sound like it's QUIC-specific, does it? Should we make this a QuicConnPool?

Actually, looking at source/extensions/upstreams/http/udp/upstream_request.h, it looks like it depends on some QUICHE capsule stuff, but doesn't depend (at least not directly) on QUIC stuff. though for some reason it seems to want to #include quic_spdy_stream.h, but it doesn't seem to use it? I'm confused :)

Reply:
@DavidSchinazi, I'm also struggling to fix the leakage from this code. Can you clarify the dependency model here?

Reply:
`UdpConnPool` was written by @jeongseokson to add CONNECT-UDP support in #27714. CONNECT-UDP works with HTTP/2 so (in theory) this should work even when HTTP/3 support is compiled out. In practice, that might require a few small refactors to QUICHE because the QUICHE code wasn't designed with the requirement to allow compiling HTTP/3 support out

Reply:
QUICHE http2 code doesn't depend on QUICHE quic code. So I *think* we could be able to draw a clear line between http2 code needed by CONNECT-UDP and quic code.

Reply:
Right, that's what I was suggesting. Looking at the *code* it doesn't seem to use QUIC and so we should not need to conditionally compile it. However, for some reason (bug) it does #include quic_spdy_stream.h. I suspect we should just remove that #include?

Reply:
I think @RyanTheOptimist is right, quic_spdy_stream.h. should have not been included in the first place. Sorry for leaving an unncessary include there. Would removing the quic_spdy_stream.h include solve the issue?